VogtObserver/Sum.cs

```diff
@@ -6,9 +6,11 @@ namespace VogtObserver
 {
     class Sum : IObserver
     {
-        public Sum()
+        private readonly RandomNumberGenerator _randomNumberGenerator;
+        public Sum(RandomNumberGenerator randomNumberGenerator)
         {
-
+            _randomNumberGenerator = randomNumberGenerator;
```

Comment:
Dont pass in RandomNumberGenerator in constructor. See my comment about not using dependency injection on the observable/observer relationship specifically. Your goal is to abstract the observer from the observable, and vice versa.